aten/src/ATen/xpu/profiler/PtiActivityHandlers.cpp

```diff
@@ -0,0 +1,350 @@
+      linkedActivity(activity->_correlation_id, cpuCorrelationMap_);
+  traceBuffer_.emplace_activity(
+      traceBuffer_.span, act_t::XPU_RUNTIME, std::string(activity->_name));
+  auto& runtime_activity = traceBuffer_.activities.back();
+  runtime_activity->startTime = activity->_start_timestamp;
+  runtime_activity->endTime = activity->_end_timestamp;
+  runtime_activity->id = activity->_correlation_id;
+  runtime_activity->device = activity->_process_id;
+  runtime_activity->resource = activity->_thread_id;
+  runtime_activity->threadId = activity->_thread_id;
+  runtime_activity->flow.id = activity->_correlation_id;
+  runtime_activity->flow.type = libkineto::kLinkAsyncCpuGpu;
+  runtime_activity->flow.start =
+      (runtime_activity->name() == "piEnqueueKernelLaunch" ? 1 : 0);
+  runtime_activity->linked = linked;
+  runtime_activity->addMetadata("correlation", activity->_correlation_id);
+
+  checkTimestampOrder(&*runtime_activity);
+  if (outOfRange(*runtime_activity)) {
+    removeCorrelatedPtiActivities(&*runtime_activity);
+    traceBuffer_.activities.pop_back();
+    return;
+  }
+  runtime_activity->log(*logger);
+}
+
+void XPUActivityProfilerSession::handleKernelActivity(
+    const pti_view_record_kernel* activity,
+    logger_t* logger) {
+  traceBuffer_.span.opCount += 1;
+  traceBuffer_.gpuOpCount += 1;
+  const itrace_t* linked =
+      linkedActivity(activity->_correlation_id, cpuCorrelationMap_);
+  traceBuffer_.emplace_activity(
+      traceBuffer_.span,
+      act_t::CONCURRENT_KERNEL,
+      std::string(activity->_name));
+  auto& kernel_activity = traceBuffer_.activities.back();
+  kernel_activity->startTime = activity->_start_timestamp;
+  kernel_activity->endTime = activity->_end_timestamp;
+  kernel_activity->id = activity->_correlation_id;
+  kernel_activity->device =
+      c10::xpu::get_device_idx_from_uuid(activity->_device_uuid);
+  kernel_activity->resource = getMappedQueueId(activity->_sycl_queue_id);
+  kernel_activity->threadId = activity->_thread_id;
+  kernel_activity->flow.id = activity->_correlation_id;
+  kernel_activity->flow.type = libkineto::kLinkAsyncCpuGpu;
+  kernel_activity->flow.start = 0;
+  kernel_activity->linked = linked;
+  kernel_activity->addMetadata("appended", activity->_append_timestamp);
+  kernel_activity->addMetadata("submitted", activity->_submit_timestamp);
+  kernel_activity->addMetadata("device", kernel_activity->deviceId());
+  kernel_activity->addMetadataQuoted(
+      "context", handleToHexString(activity->_context_handle));
+  kernel_activity->addMetadata("sycl queue", activity->_sycl_queue_id);
+  kernel_activity->addMetadataQuoted(
+      "l0 queue", handleToHexString(activity->_queue_handle));
+  kernel_activity->addMetadata("correlation", activity->_correlation_id);
+  kernel_activity->addMetadata("kernel_id", activity->_kernel_id);
+
+  checkTimestampOrder(&*kernel_activity);
+  if (outOfRange(*kernel_activity)) {
+    removeCorrelatedPtiActivities(&*kernel_activity);
```

Comment:
When the act is out of capture range, the gpu op count should be minus 1 at the same time.

Reply:
done